kafka-commons/src/test/java/uk/gov/homeoffice/digital/sas/kafka/EmbeddedKafkaIntegrationTest.java

```diff
@@ -0,0 +1,129 @@
+package uk.gov.homeoffice.digital.sas.kafka;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.kafka.test.context.EmbeddedKafka;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.annotation.DirtiesContext.ClassMode;
+import uk.gov.homeoffice.digital.sas.config.TestConfig;
+import uk.gov.homeoffice.digital.sas.kafka.consumer.KafkaConsumer;
+import uk.gov.homeoffice.digital.sas.kafka.message.KafkaAction;
+import uk.gov.homeoffice.digital.sas.kafka.producer.KafkaProducerService;
+import uk.gov.homeoffice.digital.sas.model.Profile;
+import uk.gov.homeoffice.digital.sas.repository.ProfileRepository;
+
+@SpringBootTest(classes = TestConfig.class)
+@DirtiesContext(classMode = ClassMode.AFTER_EACH_TEST_METHOD)
+@EmbeddedKafka(
+    partitions = 1,
+    brokerProperties = {
+        "listeners=PLAINTEXT://localhost:3333",
+        "port=3333"
+    }
+)
+class EmbeddedKafkaIntegrationTest {
+
+  private static final String PROFILE_ID = "profileId";
+  private static final String PROFILE_NAME = "Original profile";
+  private static final String UPDATED_PROFILE_NAME = "Updated profile";
+  private static final int CONSUMER_TIMEOUT = 10;
+  private String EXPECTED_CREATE_MESSAGE_PAYLOAD;
+  private Profile profile;
+
+  @Value("${projectVersion}")
+  private String version;
+
+  @Autowired
+  private KafkaProducerService<Profile> kafkaProducerService;
+
+  @Autowired
+  private KafkaConsumer kafkaConsumer;
+
+  @Autowired
+  private ProfileRepository profileRepository;
+
+  @BeforeEach
+  void setup() {
+    profile = new Profile(PROFILE_ID, PROFILE_NAME);
+    EXPECTED_CREATE_MESSAGE_PAYLOAD = generatePayload(version, profile, KafkaAction.CREATE);
+  }
+
+  @Test
+  void shouldSendCreateMessageToTopicFromProducer() throws Exception{
```

Comment:
might want to consider following the test name pattern used in our other libraries/services: `[methodname]_[given/when]_[then]`

Reply:
As this is an end-to-end test, it doesn't apply to a specific method. I borrowed the pattern used by Spring project tests

Reply:
Ah I was thinking the starting test point was sendMessage, but yes that makes sense to me as it is end to end. Will resolve.